source/opt/iterator.h

```diff
@@ -158,8 +160,9 @@ inline bool UptrVectorIterator<VT, IC>::operator<(
 }
 
 template <typename VT, bool IC>
-inline UptrVectorIterator<VT, IC> UptrVectorIterator<VT, IC>::InsertBefore(
-    Uptr value) {
+template <bool C>
```

Comment:
It's confusing to have different template names in the declaration and this definiton.
In the declaration we have:  IsConst as the outer template param, and IC as the inner template param.
In this definition we have: IC as the outer termplate param, and C as the inner template param.

I see the readability gain from using short names like "IC".  But the confusion comes because the same short name (IC) is used for two different things.  Please rename to avoid this kind of conflict.  E.g. `<bool MethodForConstObj = IsConst>` at the declaration, and `<bool MethodForConstObj>` at the definition.